src/databricks/labs/ucx/install.py

```diff
@@ -622,9 +654,20 @@ def _upload_wheel(self) -> str:
             remote_wheel = f"{self._install_folder}/wheels/{local_wheel.name}"
             remote_dirname = os.path.dirname(remote_wheel)
             with local_wheel.open("rb") as f:
-                self._ws.dbfs.mkdirs(remote_dirname)
-                logger.info(f"Uploading wheel to dbfs:{remote_wheel}")
-                self._ws.dbfs.upload(remote_wheel, f, overwrite=True)
+                try:
+                    self._ws.dbfs.mkdirs(remote_dirname)
+                    logger.info(f"Uploading wheel to dbfs:{remote_wheel}")
+                    self._ws.dbfs.upload(remote_wheel, f, overwrite=True)
+                except OperationFailed as err:
+                    logger.warning(f"Uploading wheel file to DBFS failed, DBFS is probably write protected. {err}")
+                    # assume DBFS is write protected
+                    self._write_protected_dbfs = True
```

Comment:
can we ask for `self._configure_override_clusters()` in this branch? it'll be silly to make installation more complex for 99% of customers, who don't have a write-protected DBFS.

Reply:
Hmmm... cluster overrides are handy for a lot of cases > 1%, but not more than 50%.

Reply:
The intended select logic will return only clusters with the correct security mode.

Reply:
Done

Reply:
Done

Reply:
@nfx The `config.yml` file is written before the upload wheel is attempted and the write protected DBFS is detected. Delaying the cluster config means that the overrides are not available when it comes time to write the config. Do we write twice, do we delay the write of the config, does that mess up other logic?